src/main/java/pl/krakow/phonebook/customer/repository/CustomerRepositoryImpl.java

```diff
@@ -0,0 +1,56 @@
+package pl.krakow.phonebook.customer.repository;
+
+import org.springframework.stereotype.Component;
+import pl.krakow.phonebook.customer.domain.Customer;
+import pl.krakow.phonebook.customer.exception.CustomerExistException;
+import pl.krakow.phonebook.customer.exception.CustomerNotFoundException;
+
+import java.util.List;
+
+@Component
+public class CustomerRepositoryImpl implements CustomerRepository {
+
+    private List<Customer> data;
+
+    public CustomerRepositoryImpl(List<Customer> data) {
+        this.data = data;
+    }
+
+    @Override
+    public List<Customer> findAll() {
+        return data;
+    }
+
+    @Override
+    public Customer findByName(String name) throws CustomerNotFoundException {
+        for (Customer customer: data) {
```

Comment:
You use Java11 in the project so usage of StreamAPI looks possible here .

Reply:
done